azkaban-hadoop-security-plugin/src/main/java/azkaban/security/HadoopSecurityManager_H_2_0.java

```diff
@@ -334,12 +334,22 @@ private CredentialProviderWithKeyStore getCustomCredentialProvider(final Props p
 
   private void registerCustomCredential(final Props props, final Credentials hadoopCred,
       final String userToProxy, final Logger jobLogger, final String customCredentialProviderName) {
-    final CredentialProviderWithKeyStore customCredential = getCustomCredentialProvider(
+    final CredentialProvider customCredential = getCustomCredentialProvider(
             props, hadoopCred, jobLogger, customCredentialProviderName);
     final KeyStore keyStore = KeyStoreManager.getInstance().getKeyStore();
     if (keyStore != null) {
       // Containerized Execution.
-      customCredential.setKeyStore(keyStore);
+      try {
+        ((CredentialProviderWithKeyStore)customCredential).setKeyStore(keyStore);
+      } catch (ClassCastException e) {
+        logger.error("Encountered error while casting to "
+                + customCredentialProviderName, e);
+        throw new IllegalStateException("Encountered error while casting to "
+                + customCredentialProviderName, e);
+      } catch (final Exception e) {
+        logger.error("Unknown error occurred while setting keyStore");
```

Comment:
Would you like to also report `e` here (and on the next line), so that we know what actually failed and why?

Reply:
Will do.